examples/stock/company.v.py

```diff
@@ -25,29 +27,23 @@ def buy_stock():
     buy_order = msg.value / self.price # rounds down
 
     # There are enough shares to buy
-    assert self.avail_shares >= buy_order
+    assert self.holdings[self.company] >= buy_order
     
     # Take the shares off the market and give to stockholder
-    self.avail_shares -= buy_order
-    if self.holdings[msg.sender]:
-        # Account exists, so add to that account
-        self.holdings[msg.sender] += buy_order
-    else:
-        # Not currently a stockholder, so add them
-        self.holdings[msg.sender] = buy_order
+    self.holdings[self.company] -= buy_order
+    self.holdings[msg.sender] += buy_order
 
 # Give stock back to company and get my money back!
-def sell_stock(sell_order: num):
+def sell_stock(sell_order: currency_value):
     # Can only sell as much stock as you own
     assert self.holdings[msg.sender] >= sell_order
 
     # Sell the stock, send the proceeds to the user
+    # and put the stock back on the market
     self.holdings[msg.sender] -= sell_order
+    self.holdings[self.company] += sell_order
     send(msg.sender, sell_order * self.price)
 
-    # Put the stock back on the market
-    self.avail_shares += sell_order
-
 @constant
 def valuation() -> wei_value:
-    return (self.total_shares - self.avail_shares) * self.price
+    return (self.total_shares - self.holdings[self.company]) * self.price
```

Comment:
I could be wrong but doesn't valuation mean the total shares that exist time the price and not just the shares that the company isn't holding? (as in just `self.total_shares * self.price`)